src/jsonpatcherproxy.js

```diff
@@ -181,6 +226,30 @@ var JSONPatcherProxy = (function() {
     }
     return this.generateProxyAtPath(root, path);
   };
+  JSONPatcherProxy.prototype._resetCachedProxiesPaths = function(root, path) {
+    /* deleting array elements could render other array elements paths incorrect, 
+    this function fixes all incorrect paths efficiently */
+    for (let key in root) {
+      if (
+        root.hasOwnProperty(key) &&
+        root[key] &&
+        typeof root[key] === 'object' &&
+        root[key]._isProxified
+      ) {
+        const distPath = path + '/' + JSONPatcherProxy.escapePathComponent(key);
+
+        // get the proxy instance (an instance is {proxy, originalObject})
+        var proxyInstance = this.proxifiedObjectsMap.get(root[key]);
+        if (proxyInstance) {
+          // get the un-proxified originalObject, we need because `set` trap passes the original object as target
+          const originalObject = proxyInstance.originalObject;
+          // update its path
+          this.objectsPathsMap.set(originalObject, distPath);
+        }
+        this._resetCachedProxiesPaths(root[key], distPath);
```

Comment:
Do I get it right, that if that node is not proxified you are trying to update proxies in descendants?